sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/JavaFieldSchemaTest.java

```diff
@@ -728,4 +730,46 @@ public void testNoCreateOptionThrows() throws NoSuchSchemaException {
         thrown.getMessage(),
         containsString("zero-argument constructor"));
   }
+
+  @Test
+  public void testSelfNestedPOJOThrows() throws NoSuchSchemaException {
+    SchemaRegistry registry = SchemaRegistry.createDefault();
+    RuntimeException thrown =
+        assertThrows(
+            RuntimeException.class,
+            () -> {
+              registry.getSchema(SelfNestedPOJO.class);
+            });
+
+    assertThat(
+        "Message should suggest not using a circular schema reference.",
+        thrown.getMessage(),
+        containsString("circular reference"));
+
```

Comment:
nit: Remove space. Similar to above, it keeps the Assert section together and is more consistent with other tests.

This and the above comment should apply to other changes.